File: 0120-triangle/0120-triangle.js
const minimumTotal = (triangle) => {
    for (let i=1; i<triangle.length; i++) {
        const lastRow = triangle[i-1];
        const thisRow = triangle[i];
        for (let j=0; j<thisRow.length; j++) {
            if (j===0) {
                triangle[i][j] += lastRow[j];
            } else if (j===thisRow.length-1) {
                triangle[i][j] += lastRow[j-1];
            } else {
                triangle[i][j] += Math.min(
                    triangle[i-1][j],
                    triangle[i-1][j-1]
                )
            }
        }
    }
    return Math.min(...triangle[triangle.length-1])
};

File: 0120-triangle/0120-triangle.py
class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        prev = [0] * (len(triangle)+1)
        for row in triangle[::-1]:
            curr = []
            for i,num in enumerate(row):
                curr.append( num + min(prev[i], prev[i+1]) )
            prev = curr
        return prev[0]